vendors/pc/boards/windows/aws_demos/application_code/aws_demo_logging.c

```diff
@@ -525,8 +525,22 @@ static void prvLoggingPrintf( uint8_t usLoggingLevel,
             xLength2 = 0;
             cPrintString[ xLength ] = '\0';
         }
+        else if ( xLength2 < ( dlMAX_PRINT_STRING_LENGTH - xLength ) )
+        {
+            /* Complete log is in buffer. */
+            xLength += xLength2;
+        }
+        else
+        {
+            /* Buffer required is larger than allocated, not complete log is in buffer. */
+            xLength = dlMAX_PRINT_STRING_LENGTH ;
+        }
 
-        xLength += xLength2;
+        /* Check if the buffer has room for "\r\n" and make it if not. */
+        if (xLength > ( dlMAX_PRINT_STRING_LENGTH - 3 ) )
+        {
+            xLength -= 3;
```

Comment:
As a safety measure to prevent underflow if the `dlMAX_PRINT_STRING_LENGTH` is less than 3 characters in length (which is unlikely), the check for `dlMAX_PRINT_STRING_LENGTH` OR `xLength` >= 3 can be added

Reply:
I dont think this is required and it will not fail here first, if its not called from task then the string itself is 4 bytes and some other prefixes before the actual log, I can put a comment that minimum 128 is required , what do you think ?